connection.js

```diff
@@ -263,9 +263,24 @@ TChannelConnection.prototype.onWriteError = function onWriteError(err) {
 TChannelConnection.prototype.onHandlerError = function onHandlerError(err) {
     var self = this;
 
-    if (err) {
-        self.resetAll(err);
+    // TODO: why is err optional?
+    if (!err) {
+        return;
```

Comment:
Cursory audit of the code shows that this method is used only to listen on error, and the error event is always emitted with an error. :dwarf: and my :axe: